lib/modules/hosts/googlemaps.js

```diff
@@ -0,0 +1,46 @@
+/* @flow */
+
+import { Host } from '../../core/host';
+
+export default new Host('googlemaps', {
+	domains: ['maps.google.ca', 'maps.google.com', 'google.co.uk', 'google.com', 'google.ca'],
+	logo: 'https://maps.google.com/favicon.ico',
+	name: 'Google Maps',
+	detect: ({ host, searchParams, pathname }) => {
+		// Only valid if we can find some coords to display
+		if (host.includes('maps.') || pathname.includes('/maps')) {
+			if (searchParams.has('ll') || searchParams.get('q')) {
+				// Handle old style maps.google urls
+				return [
+					searchParams.has('ll') ? searchParams.get('ll') : searchParams.get('q'),
+					searchParams.get('z'),
+					searchParams.has('maptype') ? searchParams.get('maptype') : 'roadmap',
+				];
+			} else {
+				// Parse new style google map urls
+				const location = pathname.split('/').find(part => (part.indexOf('@') === 0));
+				if (location) {
+					const parts = location.substring(1).split(',');
```

Comment:
Destructuring this `const [long, lat, zoom] =` (or whatever they should be called) might be slightly clearer